addons/html_builder/static/src/website_builder/plugins/options/google_map_option/google_map_option.js

```diff
@@ -0,0 +1,63 @@
+import { useRef, onWillStart, onMounted } from "@odoo/owl";
+import { BaseOptionComponent, useDomState } from "@html_builder/core/utils";
+
+/** @import { Place } from './google_map_option_plugin.js' */
+/**
+ * @typedef {Object} Props
+ * @property {function(Element, boolean):Promise<void>} loadGoogleMaps
+ * @property {function(Element, Place)} onPlaceChanged
+ */
+
+export class GoogleMapOption extends BaseOptionComponent {
+    static template = "html_builder.GoogleMapOption";
+    /** @type {Props} */
+    static props = {
+        loadGoogleMaps: { type: Function },
+        onPlaceChanged: { type: Function },
+    };
+
+    async setup() {
+        super.setup();
+        /** @type {Props} */
+        this.props;
+        this.inputRef = useRef("inputRef");
+        /** @type {{ map: Element }} */
+        this.state = useDomState(map => ({ map }));
+        onWillStart(async () => {
+            await this.props.loadGoogleMaps(this.state.map);
+        });
+        onMounted(() => {
+            this.initializeAutocomplete(this.inputRef.el);
+        });
+    }
+
+    /**
+     * Initialize Google Places API's autocompletion on the option's input.
+     *
+     * @param {Element} inputEl
+     */
+    initializeAutocomplete(inputEl) {
+        if (!this.googleMapsAutocomplete && google?.maps?.places) {
+            this.googleMapsAutocomplete = new google.maps.places.Autocomplete(
+                inputEl,
+                { types: [ "geocode" ] },
+            );
```

Comment:
How it works with the history ?

Reply:
Is there a history of stuff that's in the sidebar?

Reply:
I m not sure. What we want to do exactly with this inputEl. What happens when you type a char or you press enter ?

Reply:
It's just that Google's API asks for an Input element to initialize its autocomplete feature on. So when I call `google.maps.places.Autocomplete` with that element it attaches its feature to it and now when I type it makes places suggestions. If I type enter it closes the suggestion dropdown. If I select one of the suggestions Google triggers a "place_changed" event, which I listen to to update the state.
I checked and nothing happens in the history when I do any of these things. That seems correct to me, except for the address change which should actually trigger a step. What I should probably do is add a history step in the `commitPlace` function.

Reply:
There, I did that and it works perfectly.